examples/step-69/doc/intro.dox

```diff
@@ -0,0 +1,418 @@
+&\ \ \ \ \ \ \ \  \textbf{U}_i^{n+1} \leftarrow \textbf{U}_i^{n+1} - \frac{\tau_n}{m_i}
+ \mathbb{f}(\mathbf{U}_j^{n})\cdot \mathbf{c}_{ij} + d_{ij} \mathbf{U}_j^{n} \\
+&\ \ \ \ \textbf{end} \\
+&\ \ \ \ \texttt{scatter_updated_state} (\textbf{U}_i^{n+1}) \\
+&\textbf{end}
+@f}
+
+We note here that:
+- This "assembly" does not require any form of quadrature or cell-loops.
+- Here $\textbf{c}$ and $\textbf{U}^n$ are a global matrix and a global vector
+containing all the vectors $\mathbf{c}_{ij}$ and all the states
+$\mathbf{U}_j^n$ respectively.
+- $\texttt{gather_cij_vectors}$, $\texttt{gather_state_vectors}$, and
+$\texttt{scatter_updated_state}$ are hypothetical implementations that
+either collect (from) or write (into) global matrices and vectors.
+- Note that: if we assume a cartesian mesh in two space
+dimensions, first-order polynomial space $\mathbb{Q}^1$, and that
+$\mathbf{x}_i$ is an interior node (i.e. $\mathbf{x}_i$ is not on the boundary
+of the domain ) then: $\{\textbf{U}_j^n\}_{j \in \mathcal{I}(i)}$ should contain
+nine state vector elements (i.e. all the states in the patch/macro element
+associated to the shape function $\phi_i$). This is one of the major
+differences with the usual cell-based loop where the gather functionality
+(encoded in FEValuesBase<dim, spacedim>.get_function_values() in the case
+of deal.ii) only collects values for the local cell (just a subset of the
+patch).
+
+The actual implementation will deviate from above code in one key aspect:
+the time-step size $\tau$ has to be chosen subject to a CFL condition
+@f{align*}
+  \tau_n = c_{\text{cfl}}\,\min_{
+  i\in\mathcal{V}}\left(\frac{m_i}{-2\,d_{ii}^{n}}\right),
+@f}
+where $0<c_{\text{cfl}}\le1$ is a chosen constant. This will require to
+compute all $d_{ij}$ in a separte step prior to actually performing above
+update. The core principle remains unchanged, though: we do not loop over
+cells but rather over all edges of the sparsity graph.
+
+@note It is not uncommon to encounter such fully algebraic schemes (i.e.,
+no bilinear forms, no cell loops, and no quadrature) outside of the finite
+element community in the wider CFD community. There is a rich history of
+application of this kind of schemes, also called <i>edge-based</i> or
+<i>graph-based</i> finite element schemes (see for instance
+@cite Rainald2008 for a historical overview).
+
+<h3>Conservation properties and boundary conditions</h3>
+
+In the example considered in this tutorial step we use three different types of
+boundary conditions: essential-like boundary conditions (we prescribe a state in
+the left portion of our domain), outflow boundary conditions (also called
+"do-nothing" boundary conditions) in the right portion of the domain, and
+"reflecting boundary" conditions (also called "slip" boundary conditions)
+@f{align*}
+  \mathbf{m}_j \cdot \boldsymbol{\nu}_j \equiv 0  \ \
+  \mathbf{x}_j \in \partial\Omega \, .
```

Comment:
I don't understand what the index `j` is doing in this formula. Can you explain?

Reply:
The index "j" was removed. That entire section was quite a bit.